### DISCUSSION TITLE
about log serialize

### DISCUSSION THREAD

**Discussion by BigCatMiaoMM** [apache/skywalking#10954] [General]
![image](https://github.com/apache/skywalking/assets/38972314/7a2865ce-4e32-457e-aec3-c8ae3317bd53)

Which implementation class do I need to look for to serialize a log file into kafka format? Which method can be called to serialize the stream of logback to the format required by kafka?

{
    "timestamp":1618161813371,
    "service":"Your_ApplicationName",
    "serviceInstance":"3a5b8da5a5ba40c0b192e91b5c80f1a8@192.168.1.8",
    "layer":"GENERAL",
    "traceContext":{
        "traceId":"ddd92f52207c468e9cd03ddd107cd530.69.16181331190470001",
        "spanId":"0",
        "traceSegmentId":"ddd92f52207c468e9cd03ddd107cd530.69.16181331190470000"
    },
    "tags":{
        "data":[
            {
                "key":"level",
                "value":"INFO"
            },
            {
                "key":"logger",
                "value":"com.example.MyLogger"
            }
        ]
    },
    "body":{
        "text":{
            "text":"log message"
        }
    }
}

![image](https://github.com/apache/skywalking/assets/38972314/3fd92932-0305-4f02-b897-2cb1289d9aee)

**Comment by wu-sheng** [comment]
What is the context of this asking? Are you implementing some clients?

**Reply by BigCatMiaoMM** [reply]
yes I want to implement kafka forwarding myself without having to borrow from any component

**Reply by wu-sheng** [reply]
Check here https://github.com/apache/skywalking-data-collect-protocol/blob/master/logging/Logging.proto 

It is protobuf formating.

**Reply by BigCatMiaoMM** [reply]
ok thank u ，I need to rewrite it to use java
